spec/salesforce_bulk/job_spec.rb

```diff
@@ -0,0 +1,247 @@
+      SalesforceBulk::Connection.new(username, password, version, true)
+    }
+
+
+  # The responses are based off version 34.0
+  # https://developer.salesforce.com/docs/atlas.en-us.196.0.api_asynch.meta/api_asynch/asynch_api_jobs_close.htm
+  # The things we chose to test are based on existing usage in Clockwork, which is `query` and `update`. These underlyingly call the following tested functions.
+
+  describe '#initialize' do
+    it 'creates a new instance of the Job class' do
+      job = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      expect(job).to be_an_instance_of(SalesforceBulk::Job)
+    end
+  end
+
+  # https://developer.salesforce.com/docs/atlas.en-us.196.0.api_asynch.meta/api_asynch/asynch_api_jobs_create.htm
+  describe '#create_job' do
+    it 'can adhere to the contract with Salesforce' do
+      allow(connection).to receive(:post_xml).with(nil, "job", '<?xml version="1.0" encoding="utf-8" ?><jobInfo xmlns="http://www.force.com/2009/06/asyncapi/dataload"><operation>' + operation + '</operation><object>' + sobject + '</object><contentType>CSV</contentType></jobInfo>', xml_headers).and_return('<?xml version="1.0" encoding="UTF-8"?><jobInfo
+   xmlns="http://www.force.com/2009/06/asyncapi/dataload">
+ <id>' + job_id + '</id>
+ <operation>' + operation + '</operation>
+ <object>' + sobject + '</object>
+ <createdById>005PJ000002voWnYAI</createdById>
+ <createdDate>2025-10-17T16:42:34.000Z</createdDate>
+ <systemModstamp>2025-10-17T16:42:34.000Z</systemModstamp>
+ <state>Open</state>
+ <concurrencyMode>Parallel</concurrencyMode>
+ <contentType>CSV</contentType>
+ <numberBatchesQueued>0</numberBatchesQueued>
+ <numberBatchesInProgress>0</numberBatchesInProgress>
+ <numberBatchesCompleted>0</numberBatchesCompleted>
+ <numberBatchesFailed>0</numberBatchesFailed>
+ <numberBatchesTotal>0</numberBatchesTotal>
+ <numberRecordsProcessed>0</numberRecordsProcessed>
+ <numberRetries>0</numberRetries>
+ <apiVersion>34.0</apiVersion>
+ <numberRecordsFailed>0</numberRecordsFailed>
+ <totalProcessingTime>0</totalProcessingTime>
+ <apiActiveProcessingTime>0</apiActiveProcessingTime>
+ <apexProcessingTime>0</apexProcessingTime>
+</jobInfo>')
+
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      actual_job_id = instance.create_job
+      expect(actual_job_id).to eq(job_id)
+      expect(instance.instance_variable_get(:@@job_id)).to eq(job_id)
```

Comment:
`@@job_id` is a class variable. You'll have to use:
```suggestion
      expect(instance.class.class_variable_get(:@@job_id)).to eq(job_id)
```
BTW I saw the same issue a few more times calling `instance_variable_get(:@@batch_id)`